### ISSUE TITLE
Add `random_state` to `statsmodel` adapter

### ISSUE THREAD

**Issue by ngupta23** [alan-turing-institute/sktime#2388]
Requesting the addition of a `random_state` argument to the statsmodels adapter and passing it appropriately to underlying non-deterministic models such as ETS so that the predictions can be made reproducible. Currently, multiple runs result in different interval results for the same quantiles.

More on this below:

> Many statsmodels models are non-deterministic, e.g., as they do some random sampling inside, especially when it comes to estimating variation. Having some intrinsic randomness there is not unusual and expected.

@fkiraly You are right, it looks like in this case, the intervals are based on simulations and those simulations are not exact or repeatable. This model has multiplicative components and hence exact method is not available

https://www.statsmodels.org/stable/_modules/statsmodels/tsa/exponential_smoothing/ets.html

```python
if method is None:
    exact_available = ["ANN", "AAN", "AAdN", "ANA", "AAA", "AAdA"]
    if results.model.short_name in exact_available:
        method = "exact"
    else:
        method = "simulated"
self.method = method
```

However, they can be made repeatable by passing a `random_state` inside the sktime ets.py `_predict_interval` method. What do you think - maybe open an enhancement request for this?  

```python
def _predict_interval(self, fh, X=None, coverage=None):
    valid_indices = fh.to_absolute(self.cutoff).to_pandas()

   start, end = valid_indices[[0, -1]]
   # Maybe add random_state in get_predictions (and accept random_state in the AutoETS initialization)?
   # https://www.statsmodels.org/stable/_modules/statsmodels/tsa/exponential_smoothing/ets.html
   prediction_results = self._fitted_forecaster.get_prediction(
       start=start, end=end
   )
```

_Originally posted by @ngupta23 in https://github.com/alan-turing-institute/sktime/issues/2320#issuecomment-1080029680_

**Comment by ngupta23** [comment]
For example, in ETS, the random state can be passed to the `get_prediction` method that is called from within sktime to make the prediction intervals deterministic.

https://github.com/alan-turing-institute/sktime/pull/2320#issuecomment-1087330315

**Comment by fkiraly** [comment]
agreed! If `random_state` is not already passed properly, it should be.

**Comment by Ris-Bali** [comment]
Do we need to pass the random_state argument to the constructor of Statsmodels adapter ? and then pass that argument to the appropriate methods(methods with get_prediction) of forcasters that interface from statsnmodels ?

**Comment by Ris-Bali** [comment]
Okay  understood ,can I work on this ?

**Comment by fkiraly** [comment]
go ahead!
I would say, pick one estimator and post it here, then get started with it.
What would also help is a list of estimators that need the addition, so we can tick them off.

**Comment by Ris-Bali** [comment]
@fkiraly I checked all the statsmodel forcasters I found sktime/forecasting/theta.py to be similar to ETS but I did not see any randomness in the intervals and quantiles for thetha.py . As I have illustrated below . 

![SS](https://user-images.githubusercontent.com/81592570/162233065-e90b0291-4860-42a3-a736-f7db9688c0ce.png)

**Comment by fkiraly** [comment]
Theta, I think, is deterministic, since it computes a least squares fit.
